notebooks/221-machine-translation/221-machine-translation.ipynb

```diff
@@ -0,0 +1,286 @@
+    "\n",
+    "This model encodes sentences using the SentecePieceBPETokenizer from HuggingFace. The tokenizer vocabulary is downloaded automatically with the OMZ tool.\n",
+    "\n",
+    "#### Inputs\n",
+    "The model's input is a sequence of up to 150 tokens with the following structure: `<s>` + _tokenized sentence_ + `<s>` + `<pad>` (`<pad>` tokens pad the remaining blank spaces).\n",
+    "\n",
+    "#### Ouput\n",
+    "After the inference, we have a sequence of up to 200 tokens. The structure is the same as the one for the input."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "from openvino.runtime import Core\n",
+    "import numpy as np\n",
+    "import itertools\n",
+    "from tokenizers import SentencePieceBPETokenizer"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Downloading model\n",
+    "The following command will download the model to the current directory. Please, make sure you have run `pip install openvino-dev` beforehand."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "is_executing": true,
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "! omz_downloader --name  machine-translation-nar-en-de-0002"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Load and configure the model\n",
+    "The model is now available in the `intel/` folder. Below we load and configure its inputs and outputs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "core = Core()\n",
+    "model = core.read_model('intel/machine-translation-nar-en-de-0002/FP32/machine-translation-nar-en-de-0002.xml')\n",
+    "compiled_model = core.compile_model(model)\n",
+    "input_name = \"tokens\"\n",
+    "output_name = \"pred\"\n",
+    "model.output(output_name)\n",
+    "max_tokens = model.input(input_name).shape[1]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Load tokenizers\n",
+    "\n",
+    "NLP models usually take a list of tokens as standard input. A token is a single word converted to some integer. To provide the proper input, we need the vocabulary for such mapping. We use 'merges.txt' to infor us what sequences of letters form a token. 'vocab.json' specifies the mapping between tokens and integers.\n",
```

Comment:
"We use 'merges.txt' to infor us what sequences of letters form a token." A typo here for "infor"? Do you mean "inform"?

Reply:
Yes, it should be "inform".